packages/offix-datastore/src/storage/adapters/WebSQLAdapter.ts

```diff
@@ -0,0 +1,250 @@
+ */
+export class WebSQLAdapter implements StorageAdapter {
+  private sqlite: Database;
+  private stores: ModelSchema[] = [];
+  private dbName: string;
+  private schemaVersion: number;
+
+  constructor(dbName: string, schemaVersion: number) {
+    this.dbName = dbName;
+    this.schemaVersion = schemaVersion;
+    invariant("openDatabase" in window, "WebSQL not supported");
+    this.sqlite = window.openDatabase(
+      dbName,
+      schemaVersion.toString(),
+      "Offix datastore", 5 * 1024 * 1024
+    );
+  }
+  // TODO Wrong architecture. Store can be created on demand
+  public addStore(config: ModelSchema) {
+    this.stores.push(config);
+  }
+
+  public async createStores() {
+    const existingStoreNames = await this.getStoreNames();
+    existingStoreNames.forEach((storeName) => {
+      const existingModelStoreName = this.stores.find(((store) => (storeName === store.getName())));
+      if (existingModelStoreName) { return; }
+      this.sqlite.transaction((tx) => {
+        tx.executeSql("DROP TABLE ?", [existingModelStoreName], () => {
+          logger("Store deleted", existingModelStoreName);
+        }, errorCallback);
+      });
+    });
+
+    this.stores.forEach((store) => {
+      if (existingStoreNames.includes(store.getName())) { return; }
+      const stmt = this.getCreateStatement(store);
+      this.sqlite.transaction((tx) => {
+        tx.executeSql(stmt, [], () => logger("Store created", store), errorCallback);
+      });
+    });
+  }
+
+  public async save(storeName: string, input: any): Promise<any> {
+    const [cols, vals] = prepareStatement(input, "insert");
+    const stmt = `INSERT INTO ${storeName} ${cols}`;
+    return this.transaction(stmt, vals);
+  }
+
+  public async query(storeName: string, predicate?: PredicateFunction): Promise<any> {
+    if (!predicate) {
+      return await this.getStore(storeName);
+    }
+    const condition = predicateToSQL(predicate as ModelFieldPredicate);
+    const query = `SELECT * FROM ${storeName} ${condition}`;
+    // @ts-ignore
+    const res = await this.readTransaction(query, [predicate.value]);
+    return res;
+  }
+
+  public async update(storeName: string, input: any, predicate?: PredicateFunction): Promise<any> {
+    const condition = predicateToSQL(predicate as ModelFieldPredicate);
+    const [cols, vals] = prepareStatement(input, "update");
+    const query = `UPDATE ${storeName} SET ${cols} ${condition}`;
+    // @ts-ignore
+    return this.transaction(query, [...vals, predicate.value]);
+  }
+
+  public async remove(storeName: string, predicate?: PredicateFunction): Promise<any> {
+    const condition = predicateToSQL(predicate as ModelFieldPredicate);
+    const query = `DELETE FROM ${storeName} ${condition}`;
+    // @ts-ignore
+    return this.transaction(query, [predicate.value]);
+  }
+
+  public getSQLiteInstance() {
+    return this.sqlite;
+  }
+
+  createTransaction(): Promise<StorageAdapter> {
```

Comment:
Let's crete separate issue. We can return this for the moment

Reply:
@wtrocki I'll just leave them unimplemented for now then - if they are not there we get typescript errors all over the place.